{{ cookiecutter.repo_name }}/Makefile

```diff
@@ -21,6 +21,9 @@ ci-flake8: ## Test lint compliance using flake8. Config in tox.ini file
 ci-test:  ## Runs unit tests using pytest
 	docker exec $(CONTAINER_NAME) pytest /mnt/{{ cookiecutter.repo_name.replace("-", "_") }}
 
+ci-test-interactive:  ## Runs unit tests using pytest, and gives you an interactive IPDB session at the first failure
+	docker exec -it $(CONTAINER_NAME) pytest /mnt/{{ cookiecutter.repo_name.replace("-", "_") }}  -x --pdb --pdbcls=IPython.terminal.debugger:Pdb
```

Comment:
Question, is this the same as `pytest -s`? https://stackoverflow.com/questions/16022915/how-to-execute-ipdb-set-trace-at-will-while-running-pytest-tests

Reply:
It looks like `-s` is orthogonal to `-x` and the pdb options. I'm not completely sure, but my understanding based on [this doc](https://docs.pytest.org/en/latest/capture.html) is that:

* Pytest by default captures all stdout/stderr, and shows it only if a test fails
* Using `-s` means that there's no capturing at all, and all output gets shown (even for tests that pass)
* Using `-x` means that it stops after the first error, but it still dumps any captured output for that first failure.
* Using `--pdb --pdbcls=...` means that when it stops, it dumps you into an ipdb session for that error (it still shows all captured output).

I found the `--pdb --pdbcls=...`  flag/options from the second answer on that stack overflow question.

Reply:
Interesting. I've been able to get into a into a `ipdb` session using a `-s` and `set_trace()` in the `test_*.py` file, but not into the actual file I'm testing. Sounds like this is the correct solution. Thanks for clarifying this!